Reduce header includes in SVGElement.h

rniwa opened the pull request:
#### 9e08044604db3666fd14b94af81a46f91623e74e
<pre>
Reduce header includes in SVGElement.h
<a href="https://bugs.webkit.org/show_bug.cgi?id=292567">https://bugs.webkit.org/show_bug.cgi?id=292567</a>

Reviewed by Anne van Kesteren.

Reduced #include in SVGElement.h to reduce WebKit&apos;s build time.

Also convert SVGParsingError, SVGMarkerUnitsType, and CTMScope to enum classes.

* Source/WebCore/dom/Node.cpp:
* Source/WebCore/rendering/svg/RenderSVGResourceMarker.cpp:
(WebCore::RenderSVGResourceMarker::markerTransformation const):
* Source/WebCore/rendering/svg/legacy/LegacyRenderSVGResourceMarker.cpp:
(WebCore::LegacyRenderSVGResourceMarker::markerTransformation const):
* Source/WebCore/rendering/svg/legacy/LegacyRenderSVGResourceMarker.h:
* Source/WebCore/svg/SVGAnimationElement.h:
* Source/WebCore/svg/SVGCircleElement.cpp:
(WebCore::SVGCircleElement::attributeChanged):
* Source/WebCore/svg/SVGComponentTransferFunctionElement.h:
* Source/WebCore/svg/SVGCursorElement.cpp:
(WebCore::SVGCursorElement::attributeChanged):
* Source/WebCore/svg/SVGDescElement.cpp:
* Source/WebCore/svg/SVGElement.cpp:
(WebCore::SVGElement::SVGElement):
(WebCore::SVGElement::reportAttributeParsingError):
(WebCore::SVGElement::localCoordinateSpaceTransform const):
* Source/WebCore/svg/SVGElement.h:
(WebCore::SVGElement::detachAllProperties): Deleted.
(WebCore::SVGElement::className const): Deleted.
* Source/WebCore/svg/SVGElementInlines.h:
(WebCore::SVGElement::detachAllProperties):
(WebCore::SVGElement::className const):
* Source/WebCore/svg/SVGEllipseElement.cpp:
(WebCore::SVGEllipseElement::attributeChanged):
* Source/WebCore/svg/SVGFECompositeElement.cpp:
* Source/WebCore/svg/SVGFECompositeElement.h:
* Source/WebCore/svg/SVGFEGaussianBlurElement.cpp:
* Source/WebCore/svg/SVGFELightElement.h:
* Source/WebCore/svg/SVGFEMorphologyElement.cpp:
* Source/WebCore/svg/SVGFilterElement.cpp:
(WebCore::SVGFilterElement::attributeChanged):
* Source/WebCore/svg/SVGFilterPrimitiveStandardAttributes.cpp:
(WebCore::SVGFilterPrimitiveStandardAttributes::attributeChanged):
* Source/WebCore/svg/SVGFilterPrimitiveStandardAttributes.h:
* Source/WebCore/svg/SVGForeignObjectElement.cpp:
(WebCore::SVGForeignObjectElement::attributeChanged):
* Source/WebCore/svg/SVGGraphicsElement.cpp:
(WebCore::SVGGraphicsElement::SVGGraphicsElement):
(WebCore::SVGGraphicsElement::getCTM):
(WebCore::SVGGraphicsElement::getScreenCTM):
* Source/WebCore/svg/SVGGraphicsElement.h:
* Source/WebCore/svg/SVGImageElement.cpp:
(WebCore::SVGImageElement::attributeChanged):
* Source/WebCore/svg/SVGLengthValue.cpp:
(WebCore::SVGLengthValue::construct):
* Source/WebCore/svg/SVGLineElement.cpp:
(WebCore::SVGLineElement::attributeChanged):
* Source/WebCore/svg/SVGLinearGradientElement.cpp:
(WebCore::SVGLinearGradientElement::attributeChanged):
* Source/WebCore/svg/SVGLocatable.cpp:
(WebCore::SVGLocatable::computeCTM):
* Source/WebCore/svg/SVGLocatable.h:
(WebCore::SVGLocatable::localCoordinateSpaceTransform const):
* Source/WebCore/svg/SVGMarkerElement.cpp:
(WebCore::SVGMarkerElement::attributeChanged):
* Source/WebCore/svg/SVGMarkerElement.h:
* Source/WebCore/svg/SVGMarkerTypes.h:
(WebCore::SVGPropertyTraits&lt;SVGMarkerUnitsType&gt;::highestEnumValue):
(WebCore::SVGPropertyTraits&lt;SVGMarkerUnitsType&gt;::toString):
(WebCore::SVGPropertyTraits&lt;SVGMarkerUnitsType&gt;::fromString):
* Source/WebCore/svg/SVGMaskElement.cpp:
(WebCore::SVGMaskElement::attributeChanged):
* Source/WebCore/svg/SVGMaskElement.h:
* Source/WebCore/svg/SVGParsingError.h:
(): Deleted.
* Source/WebCore/svg/SVGPathSegImpl.h:
* Source/WebCore/svg/SVGPatternElement.cpp:
(WebCore::SVGPatternElement::attributeChanged):
(WebCore::SVGPatternElement::localCoordinateSpaceTransform const):
* Source/WebCore/svg/SVGPatternElement.h:
* Source/WebCore/svg/SVGRadialGradientElement.cpp:
(WebCore::SVGRadialGradientElement::attributeChanged):
* Source/WebCore/svg/SVGRectElement.cpp:
(WebCore::SVGRectElement::attributeChanged):
* Source/WebCore/svg/SVGSVGElement.cpp:
(WebCore::SVGSVGElement::attributeChanged):
(WebCore::SVGSVGElement::localCoordinateSpaceTransform const):
* Source/WebCore/svg/SVGSVGElement.h:
* Source/WebCore/svg/SVGStopElement.cpp:
(WebCore::SVGStopElement::SVGStopElement):
* Source/WebCore/svg/SVGStopElement.h:
* Source/WebCore/svg/SVGTests.cpp:
* Source/WebCore/svg/SVGTextContentElement.cpp:
(WebCore::SVGTextContentElement::attributeChanged):
* Source/WebCore/svg/SVGTextPathElement.cpp:
(WebCore::SVGTextPathElement::attributeChanged):
* Source/WebCore/svg/SVGTransformable.h:
* Source/WebCore/svg/SVGUseElement.cpp:
(WebCore::SVGUseElement::attributeChanged):
* Source/WebCore/svg/properties/SVGPropertyRegistry.h:

Canonical link: <a href="https://commits.webkit.org/294576@main">https://commits.webkit.org/294576@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3023f5e57153954ecea7fecb77449befce63d100

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/102272 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/21941 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/12255 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/107432 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/52908 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/104311 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/22249 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/30447 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/77820 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/34806 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/105278 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/17220 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/92321 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/58158 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/17053 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/10349 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/52266 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/86880 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/10419 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/109807 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/29404 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/21676 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/86801 "Found 1 new test failure: imported/w3c/web-platform-tests/service-workers/service-worker/redirected-response.https.html (failure)") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/29768 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/88524 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/86389 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/21994 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/31198 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/8916 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/23646 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/29332 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/34627 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/29143 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/32466 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/30702 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

annevk reviewed (approved):
Maybe also mention the additional improvements such as enum conversion in the commit message.

rniwa committed:
Reduce header includes in SVGElement.h
https://bugs.webkit.org/show_bug.cgi?id=292567

Reviewed by Anne van Kesteren.

Reduced #include in SVGElement.h to reduce WebKit's build time.

Also convert SVGParsingError, SVGMarkerUnitsType, and CTMScope to enum classes.

* Source/WebCore/dom/Node.cpp:
* Source/WebCore/rendering/svg/RenderSVGResourceMarker.cpp:
(WebCore::RenderSVGResourceMarker::markerTransformation const):
* Source/WebCore/rendering/svg/legacy/LegacyRenderSVGResourceMarker.cpp:
(WebCore::LegacyRenderSVGResourceMarker::markerTransformation const):
* Source/WebCore/rendering/svg/legacy/LegacyRenderSVGResourceMarker.h:
* Source/WebCore/svg/SVGAnimationElement.h:
* Source/WebCore/svg/SVGCircleElement.cpp:
(WebCore::SVGCircleElement::attributeChanged):
* Source/WebCore/svg/SVGComponentTransferFunctionElement.h:
* Source/WebCore/svg/SVGCursorElement.cpp:
(WebCore::SVGCursorElement::attributeChanged):
* Source/WebCore/svg/SVGDescElement.cpp:
* Source/WebCore/svg/SVGElement.cpp:
(WebCore::SVGElement::SVGElement):
(WebCore::SVGElement::reportAttributeParsingError):
(WebCore::SVGElement::localCoordinateSpaceTransform const):
* Source/WebCore/svg/SVGElement.h:
(WebCore::SVGElement::detachAllProperties): Deleted.
(WebCore::SVGElement::className const): Deleted.
* Source/WebCore/svg/SVGElementInlines.h:
(WebCore::SVGElement::detachAllProperties):
(WebCore::SVGElement::className const):
* Source/WebCore/svg/SVGEllipseElement.cpp:
(WebCore::SVGEllipseElement::attributeChanged):
* Source/WebCore/svg/SVGFECompositeElement.cpp:
* Source/WebCore/svg/SVGFECompositeElement.h:
* Source/WebCore/svg/SVGFEGaussianBlurElement.cpp:
* Source/WebCore/svg/SVGFELightElement.h:
* Source/WebCore/svg/SVGFEMorphologyElement.cpp:
* Source/WebCore/svg/SVGFilterElement.cpp:
(WebCore::SVGFilterElement::attributeChanged):
* Source/WebCore/svg/SVGFilterPrimitiveStandardAttributes.cpp:
(WebCore::SVGFilterPrimitiveStandardAttributes::attributeChanged):
* Source/WebCore/svg/SVGFilterPrimitiveStandardAttributes.h:
* Source/WebCore/svg/SVGForeignObjectElement.cpp:
(WebCore::SVGForeignObjectElement::attributeChanged):
* Source/WebCore/svg/SVGGraphicsElement.cpp:
(WebCore::SVGGraphicsElement::SVGGraphicsElement):
(WebCore::SVGGraphicsElement::getCTM):
(WebCore::SVGGraphicsElement::getScreenCTM):
* Source/WebCore/svg/SVGGraphicsElement.h:
* Source/WebCore/svg/SVGImageElement.cpp:
(WebCore::SVGImageElement::attributeChanged):
* Source/WebCore/svg/SVGLengthValue.cpp:
(WebCore::SVGLengthValue::construct):
* Source/WebCore/svg/SVGLineElement.cpp:
(WebCore::SVGLineElement::attributeChanged):
* Source/WebCore/svg/SVGLinearGradientElement.cpp:
(WebCore::SVGLinearGradientElement::attributeChanged):
* Source/WebCore/svg/SVGLocatable.cpp:
(WebCore::SVGLocatable::computeCTM):
* Source/WebCore/svg/SVGLocatable.h:
(WebCore::SVGLocatable::localCoordinateSpaceTransform const):
* Source/WebCore/svg/SVGMarkerElement.cpp:
(WebCore::SVGMarkerElement::attributeChanged):
* Source/WebCore/svg/SVGMarkerElement.h:
* Source/WebCore/svg/SVGMarkerTypes.h:
(WebCore::SVGPropertyTraits<SVGMarkerUnitsType>::highestEnumValue):
(WebCore::SVGPropertyTraits<SVGMarkerUnitsType>::toString):
(WebCore::SVGPropertyTraits<SVGMarkerUnitsType>::fromString):
* Source/WebCore/svg/SVGMaskElement.cpp:
(WebCore::SVGMaskElement::attributeChanged):
* Source/WebCore/svg/SVGMaskElement.h:
* Source/WebCore/svg/SVGParsingError.h:
(): Deleted.
* Source/WebCore/svg/SVGPathSegImpl.h:
* Source/WebCore/svg/SVGPatternElement.cpp:
(WebCore::SVGPatternElement::attributeChanged):
(WebCore::SVGPatternElement::localCoordinateSpaceTransform const):
* Source/WebCore/svg/SVGPatternElement.h:
* Source/WebCore/svg/SVGRadialGradientElement.cpp:
(WebCore::SVGRadialGradientElement::attributeChanged):
* Source/WebCore/svg/SVGRectElement.cpp:
(WebCore::SVGRectElement::attributeChanged):
* Source/WebCore/svg/SVGSVGElement.cpp:
(WebCore::SVGSVGElement::attributeChanged):
(WebCore::SVGSVGElement::localCoordinateSpaceTransform const):
* Source/WebCore/svg/SVGSVGElement.h:
* Source/WebCore/svg/SVGStopElement.cpp:
(WebCore::SVGStopElement::SVGStopElement):
* Source/WebCore/svg/SVGStopElement.h:
* Source/WebCore/svg/SVGTests.cpp:
* Source/WebCore/svg/SVGTextContentElement.cpp:
(WebCore::SVGTextContentElement::attributeChanged):
* Source/WebCore/svg/SVGTextPathElement.cpp:
(WebCore::SVGTextPathElement::attributeChanged):
* Source/WebCore/svg/SVGTransformable.h:
* Source/WebCore/svg/SVGUseElement.cpp:
(WebCore::SVGUseElement::attributeChanged):
* Source/WebCore/svg/properties/SVGPropertyRegistry.h:

Canonical link: https://commits.webkit.org/294576@main
